DOCS: Add glossaries

hiroyuki-sato opened the pull request:
## What

Document improvement. add glossaries

## Why ?

Many acronyms have been used in this project. I think new project members want to know these words' meanings.
I use a reStructured text format. Because It's sort friendly
 

## How ?

Add glossaries page.

<img width="1251" alt="glossaries" src="https://user-images.githubusercontent.com/767650/89527306-13278200-d824-11ea-8c65-2f5be4ec971c.png">

swx-jenkins3 commented:
Can one of the admins verify this patch?

shamisp commented:
ok to test

shamisp commented on docs/source/glossaries.rst:
```diff
@@ -0,0 +1,107 @@
+AMO         Atomic Memory Operation                                       
+ANL         Argonne National Laboratory                                   
+AZP         AZure Pipeline                                                
+bcopy       Byte copy                                                     
```
Should be buffer copy

shamisp commented on docs/source/glossaries.rst:
```diff
@@ -0,0 +1,107 @@
+AZP         AZure Pipeline                                                
+bcopy       Byte copy                                                     
+Bistro      Binary Instrumentation                                        
+BTL         Byte Transfer Layer                                           
```
Open MPI's Byte Transfer Layer

shamisp commented on docs/source/glossaries.rst:
```diff
@@ -0,0 +1,107 @@
+Bistro      Binary Instrumentation                                        
+BTL         Byte Transfer Layer                                           
+cm          Connection Manager                                            
+CMA         Cross Memory Attach                                           
```
Linux Kernel Cross Memory Attach

shamisp commented on docs/source/glossaries.rst:
```diff
@@ -0,0 +1,107 @@
+CQE         Completion Queue Entry(Infiniband)                            
+csmock      static analysis tools                                         
+CUDA        Compute Unified Device Architecture(NVIDIA)                   
+DC          Dynamically Connected transport(Infiniband)                   
```
Dynamically Connected transport(Mellanox/Nvidia Infiniband)  
(DC is not part of IBTA spec)

hiroyuki-sato replied:
done. modified Infiniband -> InfiniBand.

shamisp commented on docs/source/glossaries.rst:
```diff
@@ -0,0 +1,107 @@
+BTL         Byte Transfer Layer                                           
+cm          Connection Manager                                            
+CMA         Cross Memory Attach                                           
+CQ          Completion Queue(Infiniband)                                  
```
Infiniband  -> InfiniBand

hiroyuki-sato replied:
Changed all of Infiniband -> InfiniBand

shamisp commented on docs/source/glossaries.rst:
```diff
@@ -0,0 +1,107 @@
+IPC         Inter Process Communication                                   
+JUCX        Java API over UCP                                             
+KLM         A new sophisticated way of creating memory regions.(Mellanox) 
+KNEM        Kernel Nemesis                                                
```
Inria's Kernel module for process to process zero copy (https://knem.gitlabpages.inria.fr)

hiroyuki-sato replied:
done

shamisp commented on docs/source/glossaries.rst:
```diff
@@ -0,0 +1,107 @@
+md          Memory Domain                                                 
+MEMH        Memory Handle                                                 
+MLX         Mellanox Technologies                                         
+mlx5        Connect-X5 VPI                                                
```
Mellanox-5 generation of Mellanox device driver (I think all connectX-4 generation and above)

shamisp commented on docs/source/glossaries.rst:
```diff
@@ -0,0 +1,107 @@
+mm          Memory Mapper                                                 
+MP-XRQ      Multi-Packet XRQ                                              
+MPI         Message Passing INterface                                     
+MPICH       A MPI Implementation                                          
```
ANL MPI Implementation

shamisp commented on docs/source/glossaries.rst:
```diff
@@ -0,0 +1,107 @@
+WQ          Work Queue(Infiniband)                                        
+WQE         Work Queue Elements (pronounce WOOKIE)                        
+WR          Work Request                                                  
+XPMEM       cross partition memory                                        
```
Linux kernel module originally developed by SGI implementing cross process memory mapping with  zero copy (https://github.com/hjelmn/xpmem)

hiroyuki-sato replied:
I modified this. But, I think it's a little longer description.

hiroyuki-sato commented:
@shamisp Thank you for your comment! I fixed it.
I removed trailing spaces. So I attached diff without comparing white space.

```diff
diff --git a/docs/source/glossaries.rst b/docs/source/glossaries.rst
index 2acdc6c3c..be8dc1272 100644
--- a/docs/source/glossaries.rst
+++ b/docs/source/glossaries.rst
@@ -2,24 +2,24 @@
 Glossaries
 **********
 
-=========== ==============================================================
+=========== ====================================================================================================================================================
 Keyword     Description
-=========== ==============================================================
+=========== ====================================================================================================================================================
 ACK         Acknowledge
 am          Active Message
 AMO         Atomic Memory Operation
 ANL         Argonne National Laboratory
 AZP         AZure Pipeline
-bcopy       Byte copy                                                     
+bcopy       Buffered copy
 Bistro      Binary Instrumentation
-BTL         Byte Transfer Layer                                           
+BTL         Open MPI's Byte Transfer Layer
 cm          Connection Manager
-CMA         Cross Memory Attach                                           
-CQ          Completion Queue(Infiniband)                                  
-CQE         Completion Queue Entry(Infiniband)                            
+CMA         Linux Kernel Cross Memory Attach
+CQ          Completion Queue(InfiniBand)
+CQE         Completion Queue Entry(InfiniBand)
 csmock      static analysis tools
 CUDA        Compute Unified Device Architecture(NVIDIA)
-DC          Dynamically Connected transport(Infiniband)                   
+DC          Dynamically Connected transport(Mellanox/Nvidia InfiniBand)
 DPM         Direct Packet Mode
 ep          EndPoint
 FC          Flow Control
@@ -28,23 +28,23 @@ GDR         GPUDirect RDMA
 gtest       Google Test
 HPC         High Performance Computing
 HWTM        HardWare Tag Matching
-IB          Infiniband                                                    
+IB          InfiniBand
 iface       Interaface
 IPC         Inter Process Communication
 JUCX        Java API over UCP
 KLM         A new sophisticated way of creating memory regions.(Mellanox)
-KNEM        Kernel Nemesis                                                
+KNEM        Inria's Kernel module for process to process zero copy (https://knem.gitlabpages.inria.fr)
 LDPM        Legacy DPM
 LLNL        Lawrence Livermore National Laboratory
 madvise     give advice about use of memory. See manual madvise(2)
 md          Memory Domain
 MEMH        Memory Handle
 MLX         Mellanox Technologies
-mlx5        Connect-X5 VPI                                                
+mlx5        Mellanox-5 generation of Mellanox device driver
 mm          Memory Mapper
 MP-XRQ      Multi-Packet XRQ
 MPI         Message Passing INterface
-MPICH       A MPI Implementation                                          
+MPICH       ANL MPI Implementation
 MTT         The MPI Testing Tool
 NAK         Negative Acknowledge
 ODP         OnDemand Paging
@@ -61,8 +61,8 @@ PGAS        Partitioned Global Address Space
 POSIX       Portable operating system interface
 ppn         processes per node
 PR          Pull Request
-QP          Queue Pair(Infiniband)                                        
-RC          Reliable Connection (Infiniband)                              
+QP          Queue Pair(InfiniBand)
+RC          Reliable Connection (InfiniBand)
 rcache      Registration Cache
 RDMA        Remote Direct Memory Access
 REQ         Request
@@ -75,7 +75,7 @@ RTE         Run Time Environment
 RX          Receive
 Skb         Socket Buffer
 sm          Shared Memory
-SM          Subnet Manager(Infiniband)                                    
+SM          Subnet Manager(InfiniBand)
 SockCM      Socket Connection Manager
 SRQ         Shared Receive Queue
 SYSV        UNIX System V
@@ -83,7 +83,7 @@ tl          Transport Layer
 TLS         Transpot LayerS
 TM          Tag Matching
 TX          Transmit
-UC          Unreliable Connection (Infiniband)                            
+UC          Unreliable Connection (InfiniBand)
 UCC         Unified Collective Communications
 UCF         Unified Communication Framework
 UCM         Unified Communication Memory
@@ -91,17 +91,17 @@ UCP         Unified Communication Protocols Higher level API
 UCS         Unified Communication Service Common utilities.
 UCT         Unified Communication Transport Lower level API
 UCX         Unified Communication X
-UD          Unreliable Datagram (Infiniband)                              
+UD          Unreliable Datagram (InfiniBand)
 uGNI        user level generic network interface(Cray)
 UMR         User mode memory registration
 VPI         Virtual Protocol Interconnect
 WIP-DNM     Work-in-Progress-Do-Not-Merge
-WQ          Work Queue(Infiniband)                                        
+WQ          Work Queue(InfiniBand)
 WQE         Work Queue Elements (pronounce WOOKIE)
 WR          Work Request
-XPMEM       cross partition memory                                        
-XRC         eXtended Reliable Connection(Infiniband)                      
+XPMEM       `XPMEM <https://github.com/hjelmn/xpmem>`_ Linux kernel module originally developed by SGI implementing cross process memory mapping with zero copy
+XRC         eXtended Reliable Connection(InfiniBand)
 XRCD        XRC Domains
 XRQ         Mellanox Shared Receive Queue
 Zcopy       Zero Copy
-=========== ==============================================================
+=========== ====================================================================================================================================================
```

hiroyuki-sato commented:
Could someone rerun CI? This PR modify document only. So `gtest` will should complete.

brminich commented:
/azp run

yosefe commented on docs/source/glossaries.rst:
```diff
@@ -0,0 +1,107 @@
+Keyword     Description
+=========== ====================================================================================================================================================
+ACK         Acknowledge
+am          Active Message
```
AM (pls use upper case for all acronyms)

hiroyuki-sato replied:
done. I modified some words to uppercase. Please take a look again.

yosefe commented on docs/source/glossaries.rst:
```diff
@@ -0,0 +1,107 @@
+am          Active Message
+AMO         Atomic Memory Operation
+ANL         Argonne National Laboratory
+AZP         AZure Pipeline
```
Azure

yosefe commented on docs/source/glossaries.rst:
```diff
@@ -0,0 +1,107 @@
+iface       Interaface
+IPC         Inter Process Communication
+JUCX        Java API over UCP
+KLM         A new sophisticated way of creating memory regions.(Mellanox)
```
remove `.`

yosefe commented on docs/source/glossaries.rst:
```diff
@@ -0,0 +1,107 @@
+KNEM        Inria's Kernel module for process to process zero copy (https://knem.gitlabpages.inria.fr)
+LDPM        Legacy DPM
+LLNL        Lawrence Livermore National Laboratory
+madvise     give advice about use of memory. See manual madvise(2)
```
` See madvise (2)`

yosefe commented on docs/source/glossaries.rst:
```diff
@@ -0,0 +1,107 @@
+ODP         OnDemand Paging
+OFA         OpenFabrics Alliance
+OMPI        OpenMPI
+OOB         Out of band
```
Out of band / Out of box

yosefe commented on docs/source/glossaries.rst:
```diff
@@ -0,0 +1,107 @@
+Open MPI    A MPI Implementation
+ORNL        Oak Ridge National Laboratory
+PCIe        PCI Express
+PFN         Physical Page Frame
```
Physical Frame Number

yosefe commented on docs/source/glossaries.rst:
```diff
@@ -0,0 +1,107 @@
+REQ         Request
+rkey        Remote KEY
+RMA         Remote Memory Access
+RNR         Receiver Not Ready
```
RnR

yosefe commented on docs/source/glossaries.rst:
```diff
@@ -0,0 +1,107 @@
+ROCm        Radeon Open Compute platform(AMD)
+RTE         Run Time Environment
+RX          Receive
+Skb         Socket Buffer
```
skb

yosefe commented on docs/source/glossaries.rst:
```diff
@@ -0,0 +1,107 @@
+SM          Subnet Manager(InfiniBand)
+SockCM      Socket Connection Manager
+SRQ         Shared Receive Queue
+SYSV        UNIX System V
```
SysV

yosefe commented on docs/source/glossaries.rst:
```diff
@@ -0,0 +1,107 @@
+UCF         Unified Communication Framework
+UCM         Unified Communication Memory
+UCP         Unified Communication Protocols Higher level API
+UCS         Unified Communication Service Common utilities.
```
remove `.`

hiroyuki-sato commented:
Hello, @yosefe Thank you for your review. Please take a look again when you get a chance.

hiroyuki-sato commented:
Thank you!
